src/Uno.UI/UI/Xaml/Controls/TreeView/TreeView.cs

```diff
@@ -0,0 +1,358 @@
+			//					}
+			//				}
+			//			}
+			//			return static_cast<winrt::IInspectable>(nullptr);
+			//		} ();
+
+			//		const auto treeViewArgs = winrt::make_self<TreeViewDragItemsCompletedEventArgs>(args, newParent);
+			//		m_dragItemsCompletedEventSource(*this, *treeViewArgs);
+		}
+
+
+		private void UpdateItemsSelectionMode(bool isMultiSelect)
+		{
+			var listControl = ListControl;
+			listControl.EnableMultiselect(isMultiSelect);
+
+			var viewModel = listControl.ListViewModel;
+			int size = viewModel.Size;
+
+			for (int i = 0; i < size; i++)
+			{
+				var updateContainer = listControl.ContainerFromIndex(i) as TreeViewItem;
+				if (updateContainer != null)
+				{
+					if (isMultiSelect)
+					{
+						var targetNode = viewModel.GetNodeAt(i);
+						if (targetNode != null)
+						{
+							if (viewModel.IsNodeSelected(targetNode))
+							{
+								VisualStateManager.GoToState(updateContainer, "TreeViewMultiSelectEnabledSelected", false);
+							}
+							else
+							{
+								VisualStateManager.GoToState(updateContainer, "TreeViewMultiSelectEnabledUnselected", false);
+							}
+						}
+					}
+					else
+					{
+						VisualStateManager.GoToState(updateContainer, "TreeViewMultiSelectDisabled", false);
+					}
+				}
+			}
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			ListControl = GetTemplateChild(c_listControlName);
+			if (ListControl != null)
+			{
+				var listPtr = m_listControl;
+				var viewModel = listPtr.ListViewModel;
+				if (m_rootNode == null)
+				{
+					m_rootNode = new TreeViewNode();
+				}
+
+				if (ItemsSource != null)
+				{
+					viewModel.IsContentMode = true;
+				}
+				viewModel.PrepareView(m_rootNode);
+				viewModel.SetOwningList(ListControl);
+				viewModel.NodeExpanding({ this, &TreeView::OnNodeExpanding });
+				viewModel.NodeCollapsed({ this, &TreeView::OnNodeCollapsed });
+
+				var selectionMode = SelectionMode;
+				if (selectionMode == TreeViewSelectionMode.Single)
+				{
+					ListControl.SelectionMode = ListViewSelectionMode.Single;
+				}
+				else
+				{
+					ListControl.SelectionMode = ListViewSelectionMode.None;
+					if (selectionMode == TreeViewSelectionMode.Multiple)
+					{
+						UpdateItemsSelectionMode(true);
+					}
+				}
+
+				m_itemClickRevoker = listControl.ItemClick(winrt::auto_revoke, { this, &TreeView::OnItemClick });
+				m_containerContentChangingRevoker = listControl.ContainerContentChanging(winrt::auto_revoke, { this, &TreeView::OnContainerContentChanging });
+				m_dragItemsStartingRevoker = listControl.DragItemsStarting(winrt::auto_revoke, { this, &TreeView::OnListControlDragItemsStarting });
+				m_dragItemsCompletedRevoker = listControl.DragItemsCompleted(winrt::auto_revoke, { this, &TreeView::OnListControlDragItemsCompleted });
+
+				if (m_pendingSelectedNodes && m_pendingSelectedNodes.get().Size() > 0)
+				{
+					auto selectedNodes = viewModel->GetSelectedNodes();
+					for (auto const&node : m_pendingSelectedNodes.get())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)